packages/synapse-react-client/package.json

```diff
@@ -58,6 +58,7 @@
     "downshift": "^6.1.12",
     "history": "^5.3.0",
     "immutable": "4.1.0",
+    "jotai": "^2.3.1",
```

Comment:
[Jotai](https://jotai.org/) is an atomic state library that we can use to get better performance than React Context for frequently-changing state.

Reply:
I chose Jotai over [Zustand](https://github.com/pmndrs/zustand) because it's a more natural shift from React Context (in that state is tied to a component tree vs a global store).

I chose Jotai over [Recoil](https://github.com/facebookexperimental/Recoil), which has a similar API, because Meta brands Recoil as "experimental" so it seems fairly risky to choose it.

Either of these libraries (and more, including [Redux](https://redux.js.org/), [Valtio](https://github.com/pmndrs/valtio), [MobX](https://mobx.js.org/README.html), I could go on...), could have been used to get the same performance benefits. I don't feel that we're locked-in to Jotai compared to any other option, but we may want to re-evaluate our decision later if we feel we need to keep relying on these tools.